hcl/hcl.go

```diff
@@ -109,6 +108,9 @@ type GenFileBlock struct {
 	Content *hclsyntax.Attribute
 }
 
+// PartialEvaluator represents an HCL partial evaluator
+type PartialEvaluator func(hclsyntax.Expression) (hclwrite.Tokens, error)
```

Comment:
Introduced this indirection to avoid dependency cycles. It would be odd for something general purpose like hcl to depend on a stack.EvalCtx anyway + I really wanted to avoid allowing the inner evaluation context on stack.EvalCtx to leak publicly, so it seemed like the only way.